### ISSUE TITLE
Support for SERIAL type

### ISSUE THREAD

**Issue by KES777** [citusdata/citus#511]
When table is created on the master node of citus claster

```
# CREATE TABLE "offers" (
(#   "id" serial NOT NULL,
(#   "ctime" timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL,
(# );
```

We may distribute it:

```
# select master_create_distributed_table( 'offers', 'id', 'range' );
 master_create_distributed_table 
---------------------------------

(1 row)
```

But when we try to create empty shard we get error:

```
# select master_create_empty_shard( 'offers' );
WARNING:  could not receive query results from citus-work1:9701
DETAIL:  Client error: relation "offers_id_seq" does not exist
WARNING:  could not create shard on "citus-work1:9701"
WARNING:  could not receive query results from citus-work2:9701
DETAIL:  Client error: relation "offers_id_seq" does not exist
WARNING:  could not create shard on "citus-work2:9701"
ERROR:  could only create 0 of 2 of required shard replicas

```

That is because of the `id` column:

```
id               | integer                     | not null default nextval('offers_id_seq'::regclass)
```

Do not allow to distribute tables with serial type.

**Comment by KES777** [comment]
Ambiguous because it is not clear: the tables with serial type not supported or master_create_empty_shard does not distribute required info.

**Comment by mtuncer** [comment]
Hey @KES777 

This issue is related to serial type not being supported. We have 2 issues related to this one
https://github.com/citusdata/citus/issues/211 is about the issue you are describing, https://github.com/citusdata/citus/issues/213 is about a feature that offers a fix for this one too.

**Comment by lithp** [comment]
Unfortunately the fix for #213 won't cover this. Even though nextval() will be supported, when we try to create shards the worker fails on `not null default nextval('offers_id_seq'::regclass)`.

Fixes:
1. Also push down sequence creation so the workers get their own (unused) sequences
2. Notice SERIAL and emit modified DML commands to the workers, they only see an `not null` type

**Comment by marcocitus** [comment]
Serial can now be used in distributed tables, provided that the user also creates the sequence on all the workers and that it is not the partition column. If the sequences do not exist on the worker, master_create_worker_shards errors out because it does not recognize the sequence reference in the create table command (e.g. "id int not null default nextval('events_id_seq')"), but the sequences on the workers are never actually used, since nextval is evaluated by the master. If the partition column is serial, INSERT will error out, though COPY will work.

A way of addressing the master_create_worker_shards error would be to strip the default value from the shards. When we do an INSERT or COPY, we send all columns with the default filled in even if the user only provided a subset, so there is no need for the workers to know about default values.

**Comment by ozgune** [comment]
@marcocitus -- I'm sifting through our old issues and had a question on this one.

I think we can now create a distributed table that has a ``SERIAL`` column, without asking the user to create sequences on workers. Does that align with your understanding as well? If so, I'm planning to close this issue.

(We have a separate issue related to using a ``SERIAL`` column for distribution in #687. We're keeping that one open.)

**Comment by marcocitus** [comment]
Yep, fixed by #673
